### PR TITLE
tools:list-version

### PR DESCRIPTION
**Pull request by dllgzcl**
Tools:add feature about list version of map。
Use this tool, we can list the first/last committed version of maps.

### CODE REVIEW

**Comment by dllgzcl** [comment]
Would you  merge my commit into ceph:master?@smithfarm

File: src/tools/ceph_monstore_tool.cc [hunk 1]
```diff
@@ -620,7 +622,39 @@ int main(int argc, char **argv) {
                 << " version " << v << " to " << outpath
                 << std::endl;
     }
-  } else if (cmd == "dump-paxos") {
+ }else if (cmd == "list-version") {
```
Comment (liewegas) [inline src/tools/ceph_monstore_tool.cc]: --versions or --show-versions?  --list-version suggests to me that it'll list out stored versions (and not the ends of the range).

Comment (liewegas) [inline src/tools/ceph_monstore_tool.cc]: Please fix up the whitespace (e.g., space after })... see CodingStyle

File: src/tools/ceph_monstore_tool.cc [hunk 2]
```diff
@@ -620,7 +622,39 @@ int main(int argc, char **argv) {
...
+      goto done;
+    }
+    
+    unsigned int v_first = 0;
+    unsigned int v_last = 0;
+    v_first = st.get(map_type,"first_committed");
+    v_last = st.get(map_type,"last_committed");
+    
+    std::cout << "first committed\t:" << v_first << "\r\n"
+              << "last  committed\t:" << v_last << std::endl;
```
Comment (liewegas) [inline src/tools/ceph_monstore_tool.cc]: :\t not \t:

**Comment by smithfarm** [comment]
@jecluis Could you take a look?

**Comment by dllgzcl** [comment]
Would you please take a look and merge it?@jecluis

**Comment by dllgzcl** [comment]
Would you please take a look and merge it?@jecluis

**Comment by dllgzcl** [comment]
Would you please take a look ?@jecluis

**Comment by dllgzcl** [comment]
There are some problems with my master, so I pull a request again!

**Comment by jecluis** [comment]
couple of coding style nits, but looks okay.

**Comment by jecluis** [comment]
also, suggested commit title: `tools: monstore: add 'show-versions' command`.

**Comment by dllgzcl** [comment]
There are some problems with my master, so I pull a request again!  ^.^ https://github.com/ceph/ceph/pull/7073

**Pull request closed without merging**